src/store/messages/saga.ts

```diff
@@ -624,8 +624,11 @@ const inProgress = {};
 export function* loadAttachmentDetails(action) {
   const { media, messageId } = action.payload;
 
-  // Check conditions to skip processing
-  if (inProgress[messageId] || media.url || media.downloadStatus === MediaDownloadStatus.Failed) {
+  if (
+    inProgress[messageId] ||
+    (media.url && !media.url.startsWith('mxc://')) ||
```

Comment:
I am assuming, when you download the media, you're considering that the media url is updated from `mxc://` -> `blob://..`? Which is why you've added the `mxc://` condition here?

Reply:
Yeah thats it. The matrix url is transformed into a blob, so this will exit early and not process again